src/util/adb.js

```diff
@@ -210,10 +210,45 @@ export default class ADBUtils {
     await this.runShellCommand(deviceId, ['mkdir', '-p', artifactsDir]);
 
     this.artifactsDirMap.set(deviceId, artifactsDir);
-
     return artifactsDir;
   }
 
+  async checkOrCleanArtifacts(
+    deviceId: string, clean: boolean): Promise<boolean> {
+    const {adbClient} = this;
+
+    var directories = false;
+
+    await wrapADBCall(async () => {
+      log.debug('Finding Older Artifacts');
+
+      const dirs = await adbClient.readdir(deviceId, '/sdcard');
+      for (var dir = dirs.length - 1; dir >= 0; --dir ) {
+        if (!dirs[dir].isFile() &&
```

Comment:
I thin I would prefer if we would use `file.isDirectory()`, because it is a bit more explicit, and we are only interested in directories (everything else, like symbolic links, device or socket files etc. does not interest us here)